src/init.cpp

```diff
@@ -135,7 +135,7 @@ static const char* BITCOIN_PID_FILENAME = "bitcoind.pid";
 
 static fs::path GetPidFile(const ArgsManager& args)
 {
-    return AbsPathForConfigVal(fs::PathFromString(args.GetArg("-pid", BITCOIN_PID_FILENAME)));
+    return AbsPathForConfigVal(args.GetPathArg("-pid", BITCOIN_PID_FILENAME));
```

Comment:
693cdca5f552f74de3665a874ce6acf6c48cb555

```suggestion
    return AbsPathForConfigVal(args.GetPathArg("-pid", fs::path{BITCOIN_PID_FILENAME}));
```

Reply:
Is similar reasoning applicable to 5b946edd73640c6ecdfb4cbac1d4351e634678dc _"util, refactor: Use GetPathArg to read "-settings" value"_: https://github.com/bitcoin/bitcoin/blob/5b946edd73640c6ecdfb4cbac1d4351e634678dc/src/util/system.cpp#L522?

Reply:
> Is similar reasoning applicable to [5b946ed](https://github.com/bitcoin/bitcoin/commit/5b946edd73640c6ecdfb4cbac1d4351e634678dc) _"util, refactor: Use GetPathArg to read "-settings" value"_:

Yes thanks for pointing that out. Since this PR was merged already, I wouldn't open a separate PR just to simplify that line, but there should be other chances to clean it up in the future

Reply:
Addressed as a part of #24675.